services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -42,73 +42,45 @@ async function getDefaultSchema(connection, userProvidedSchema) {
-      // Special case for json/jsonb
-      const [, content] = expression.match(/^'(.*)'::jsonb?$/i);
-      result = JSON.parse(content);
-    } else if (/^'.*'::[a-z ]+$/i.test(expression)) {
-      // Catches types containing only alpha and spaces (int, varchar, timestamp with timezone, ...)
-      // This excludes arrays or other compound types (int[], ...).
-      const [, content] = expression.match(/^'(.*)'::[a-z ]+$/i);
-      result = content;
-    } else {
-      throw new Error('SQL Expression does not match any pattern');
-    }
-  } catch (e) {
-    result = Sequelize.literal(expression);
-  }
-
-  return result;
-}
-
 /** Retrieve the description of the fields in a given table. */
 async function analyzeFields(queryInterface, tableName, config) {
-  // There is a bug in sequelize/dialects/mysql/query-generator#describe
+  const dialect = queryInterface.sequelize.getDialect();
+  let columnsByName;
+
+  // Workaround bug in sequelize/dialects/mysql/query-generator#describe
   // Sequelize queries over `[schema].[table]` instead of `[schema]`.`[table]`
   // => Don't provide the schema when using mysql/mariadb
-  const dialect = queryInterface.sequelize.getDialect();
-  const workaroundConfig = ['mysql', 'mariadb'].includes(dialect) ? {} : { schema: config.dbSchema };
-  const columnsByName = await queryInterface.describeTable(tableName, workaroundConfig);
+  if (['mysql', 'mariadb'].includes(dialect)) {
+    columnsByName = await queryInterface.describeTable(tableName, {});
+  } else {
+    columnsByName = await queryInterface.describeTable(tableName, { schema: config.dbSchema });
+  }
 
-  // There is a bug when parsing default values sequelize/dialects/postgres/query.js#run()
+  // Workaround bug in sequelize/dialects/postgres/query.js#run()
   // Sequelize simply remove quotes and truncates after '::' to convert SQL expressions to JS.
   // => Fetch the real default value from the information schema
-  const getDefaultsQuery = `
-    SELECT column_name as colname, column_default as coldefault
-    FROM information_schema.columns
-    WHERE table_schema = ? AND table_name = ?
-  `;
-
-  const rows = await queryInterface.sequelize.query(getDefaultsQuery, {
-    type: queryInterface.sequelize.QueryTypes.SELECT,
-    replacements: [config.dbSchema, tableName],
-  });
+  if (dialect === 'postgres' || dialect === 'mssql') {
+    const getDefaultsQuery = `
+      SELECT column_name as colname, column_default as coldefault
+      FROM information_schema.columns
+      WHERE table_schema = ? AND table_name = ?
+    `;
+
+    const rows = await queryInterface.sequelize.query(getDefaultsQuery, {
+      type: queryInterface.sequelize.QueryTypes.SELECT,
+      replacements: [config.dbSchema, tableName],
+    });
+
+    rows.forEach((row) => {
+      columnsByName[row.colname].defaultValue = row.coldefault;
+    });
+  }
+
+  Object.values(columnsByName).forEach((column) => {
+    const myColumn = column; // Workaround pedantic linter.
```

Comment:
Because getting defaults values is buggy in MsSQL and Postgres, but not for MySQL.
=> I only run the workaround with those databases, but still need to parse the default values for MySQL